### PR TITLE
Implement duplication of seeds

### LINKED ISSUE #321: Duplicate seed data
## User story
As a seed creator user,
I want to be able to be able to duplicate a seed
So that I don't have to start from scratch when creating new seeds

## Description
Duplicate seed functionality.

## Expected behaviour/Acceptance criteria
- Button to duplicate is in edit, delete view block of buttons.
- Button label is `duplicate`
- Name of the duplicated seed is `COPY OF [copied seed name]`
- User has option of duplicated associated documents or leaving them out.
- A new folder for associated documents is created with each new seed.

### PR DESCRIPTION
**Pull request by viniciusarre**
Closes #321

### CODE REVIEW

**Comment by viniciusarre** [comment]
@bryan-gilbert This is my first attempt on fixing this. I am not sure about some of the topics mentioned on the issue. I am not sure how to fulfil this acceptance criteria: "A new folder for associated documents is created with each new seed." Also, does this checkbox I added fulfil the criteria of "User has option of duplicated associated documents or leaving them out." (I interpreted this in a literal sense, having the documents in ehrData as the elements. But, I guess this doesn't make much sense in comparison to the other criteria). Anyway, please, let me know your thoughts on it. Thank you!

File: client/src/outside/components/EhrSeedDataList.vue [hunk 1]
```diff
@@ -28,32 +28,54 @@
...
+      @cancel="cancelSeedDuplication",
+      set-footer
+      )
+      div(slot="extra", style="margin-top: calc(40vh - 25vh);") 
+        input(
+          class="checkbox",
+          type="checkbox",
+          v-model="duplicateDocuments"
+        )
+        span Add associated documents?
```
Comment (bryan-gilbert) [inline client/src/outside/components/EhrSeedDataList.vue]: Apologies for not seeing this sooner. We do not need this extra section. It is meant for the future when each seed has documents. For now, seeds only have the EHR data which needs to be copied.
Please remove this extra section. Lines 44 - 50

File: client/src/app/ui/UiConfirm.vue [hunk 2]
```diff
@@ -10,6 +10,7 @@
       )
       h2(slot="header") {{ title }}
       div(slot="body") {{ text }}
+        slot(name="extra")
```
Comment (bryan-gilbert) [inline client/src/app/ui/UiConfirm.vue]: remove line 13 .. see comment below

File: client/src/outside/components/EhrSeedDataList.vue [hunk 3]
```diff
@@ -62,6 +84,8 @@ export default {
       dialogHeader: '',
       actionType: '',
       seedId: '',
+      duplicatingSeed: {},
+      duplicateDocuments: true,
```
Comment (bryan-gilbert) [inline client/src/outside/components/EhrSeedDataList.vue]: remove line 80 duplicateDocuments

File: client/src/outside/components/EhrSeedDataList.vue [hunk 4]
```diff
@@ -140,8 +164,35 @@ export default {
...
+      this.duplicatingSeed = sv
+      this.$refs.confirmDialog.showDialog(
+        DUPLICATE.TITLE(sv.name), 
+        DUPLICATE.DESCRIPTION, 
+        'Confirm'
+      )
+    },
+    async confirmSeedDuplication () {
+      const docs = this.duplicateDocuments ? 
+        this.duplicatingSeed.ehrData : {}
```
Comment (bryan-gilbert) [inline client/src/outside/components/EhrSeedDataList.vue]: Remove 180-181 ... copy the ehr data

File: client/src/outside/components/EhrSeedDataList.vue [hunk 5]
```diff
@@ -140,8 +164,35 @@ export default {
...
+          lastUpdateDate: new Date()
+        })
+      await StoreHelper.createSeed(this, seed)
+    },
+    cancelSeedDuplication () {
+      this.duplicatingSeed = {}
+      this.duplicateDocuments = true
     }
-  }
+  },
```
Comment (bryan-gilbert) [inline client/src/outside/components/EhrSeedDataList.vue]: remove trailing comma

File: client/src/outside/components/EhrSeedDataList.vue [hunk 6]
```diff
@@ -28,32 +28,54 @@
...
+import UiConfirm from '../../app/ui/UiConfirm.vue'
 import EhrSeedDataDialog from './EhrSeedDataDialog'
 import StoreHelper from '../../helpers/store-helper'
 import EventBus from '../../helpers/event-bus'
 import { downloadSeedToFile, downObjectToFile } from '../../helpers/ehr-utils'
 import { PAGE_DATA_REFRESH_EVENT } from '../../helpers/event-bus'
 
+const DUPLICATE = {
+  TITLE : (name) => `Confirm duplication of ${name}`,
+  DESCRIPTION: 'Are you sure you\'d like to duplicate the given seed?',
```
Comment (bryan-gilbert) [inline client/src/outside/components/EhrSeedDataList.vue]: Change description to
```
  DESCRIPTION: (name) => `Are you sure you want to duplicate ${name}?`,
```
Pass in name when used below

**Pull request merged**